src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(arg) {
+    if (arg instanceof Herbivore && !arg.hidden) {
+      arg.health -= 50;
+
+      Animal.alive = Animal.alive.filter((a) => a.health > 0);
+    }
+  }
```

Comment:
The `bite` method currently only checks if the `arg` is an instance of Herbivore and not hidden before applying damage. According to the task description, it should also check if the `arg` is not an instance of Carnivore. Additionally, the method should not decrease health if the target is another Carnivore. Consider adding this condition to ensure the method behaves as expected.